python/tests/integration/arcticdb/version_store/test_basic_version_store.py

```diff
@@ -1862,3 +1862,68 @@ def test_get_existing_columns_in_series(lmdb_version_store, sym):
     assert not lmdb_version_store.read(sym, columns=["col1", "col2"]).data.empty
     if __name__ == "__main__":
         pytest.main()
+
+
+def test_use_previous_on_failure_single(lmdb_version_store):
+    lib = lmdb_version_store
+    idx = pd.date_range("2022-01-01", periods=10, freq="D")
+    l = len(idx)
+    df1 = pd.DataFrame({"a": range(l), "b": range(1, l + 1), "c": range(2, l + 2)}, index=idx)
+
+    lib.write("symbol", df1)
+
+    v1_write_time = pd.Timestamp.now()
+    time.sleep(1)
+    df2 = pd.DataFrame({"d": range(1, l + 1), "e": range(2, l + 2), "f": range(3, l + 3)}, index=idx)
+    lib.write("symbol", df2)
+
+    lib_tool = lmdb_version_store.library_tool()
+    version_keys = lib_tool.find_keys_for_id(KeyType.VERSION, "symbol")
+    assert len(version_keys) == 2
+    version_keys.sort(key=lambda k: k.creation_ts)
+
+    lib_tool.remove(version_keys[1])
+    version_keys = lib_tool.find_keys_for_id(KeyType.VERSION, "symbol")
+    assert len(version_keys) == 1
+    assert version_keys[0].version_id == 0
+
+    vit = lib.read("symbol", read_previous_on_failure=True, as_of=pd.Timestamp(v1_write_time))
```

Comment:
I don't think `failure` is the right name.

It won't `read_previous` if the data key is missing or corrupted and I can easily see a user believing that it would based on the name. 

I'm not sure what the right approach is here though. I don't think it's trivial.

Reply:
API change. **Make it clear in the PR description.**

Reply:
Let's talk about it in the next PR